### PR TITLE
Docs: add/update frontmatter descriptions for panel config pages 

### PR DESCRIPTION
**Pull request by imatwawana**
Adding concise descriptions for each page in the panel config documentation
This PR must be merged by December 15th

### CODE REVIEW

File: docs/sources/panels-visualizations/configure-data-links/index.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ labels:
     - oss
 menuTitle: Configure data links
 title: Configure data links
+description: Create data links and data link variables to create links between panels
```
Comment (nmarrs) [inline docs/sources/panels-visualizations/configure-data-links/index.md:23]: minor nit: 
```suggestion
description: Configure data links to create links between dashboards and link to external resources
```

Maybe we can also include info on that data links can be configured with template variables / panel data embedded but 1 sentence description might not be sufficient to also express this 😬

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ labels:
     - enterprise
     - oss
 title: Transform data
+description: Use transformations to manipulate your data before visualizations are applied
```
Comment (nmarrs) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:42]: maybe?
```suggestion
description: Use transformations to manipulate your data before being applied in visualizations
```

**Review by nmarrs — APPROVED** [review]
overall LGTM - just a few minor nits :)

File: scripts/docs/generate-transformations.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ labels:
     - enterprise
     - oss
 title: Transform data
+description: Use transformations to manipulate your data before visualizations are applied
```
Comment (baldm0mma) [inline scripts/docs/generate-transformations.ts:48]: @imatwawana this template builds the content in `docs/sources/panels-visualizations/query-transform-data/transform-data/index.md`, so I'm also updating here. :)

**Review by baldm0mma — APPROVED** [review]
LGTM

File: scripts/docs/generate-transformations.ts (hunk 3 above)
Comment (baldm0mma) [reply scripts/docs/generate-transformations.ts:48]: And if you update with @nmarrs suggestions, please also update this.

File: docs/sources/panels-visualizations/configure-data-links/index.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/panels-visualizations/configure-data-links/index.md:23]: Yeah these are meant to be pretty concise, but I'll make the edit you've suggested

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md (hunk 2 above)
Comment (imatwawana) [reply docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:42]: Not sure. I based it on the first few sentences of the body text. Grafana applies the transformation to the data before it applies the visualization. That's what this is meant to say. But we can also just take that part out entirely.
```suggestion
description: Use transformations to manipulate your data
```

Comment (imatwawana) [reply docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:42]: Another option
```suggestion
description: Use transformations to rename fields, join series data, apply mathematical operations, and more
```

File: scripts/docs/generate-transformations.ts (hunk 3 above)
Comment (imatwawana) [reply scripts/docs/generate-transformations.ts:48]: I wasn't sure about whether I needed to use this file to update frontmatter. Now I know, lol

Comment (baldm0mma) [reply scripts/docs/generate-transformations.ts:48]: No worries! :)

**Pull request merged**